tasks/compile-javascripts.mjs

```diff
@@ -84,6 +90,32 @@ export async function compileJavaScript ([modulePath, { srcPath, destPath }]) {
   ])
 }
 
+/**
+ * Minify JavaScript ESM to CommonJS helper
+ *
+ * @param {string} modulePath - Relative path to module
+ * @param {object} result - Generated bundle
+ * @param {string} result.code - Source code
+ * @param {import('magic-string').SourceMap} result.map - Source map
```

Comment:
**nitpick** Looks like we're only importing `magic-string` to get that type. As we're looser with type checking on the tasks, couldn't we do without it and use `any` for now waiting for an upgrade to newer versions of rollup to bring their types (which hopefully they do). Feels like we're forcing a modern approach (typing) on legacy tools (rollup 0.56.x).

**thought** Wondering if there'd be an alternative route that'd save the typing by porting the task to a `rollup.config.js` file and using `rollup-plugin-uglify` and using Rollup's CLI directly? But maybe this has constraints due to our old version and some performance downsides (in case some stuff runs in parallel a bit in the current way?).

Reply:
Yeah it was a safety thing from me

1. Rollup uses `SourceMap` via [`magic-string`](https://github.com/rollup/rollup/blob/v0.56.5/src/rollup/index.ts#LL14-L14C41)
2. UglifyJS uses plain `string` maps
3. Terser uses `EncodedSourceMap` via [`@jridgewell/gen-mapping`](https://github.com/jridgewell/gen-mapping/blob/main/src/types.ts#L12)

Wanted to be confident that the source map object had a `toString()` method

I'd love a Rollup config but thought we'd best wait for Rollup v3

Reply:
They were pretty advanced even for their old release (see [**TypeScript declarations**](https://github.com/rollup/rollupjs.org/blob/890687b8556b6134a0389284fb5fde71be8206a4/guide/en/04-javascript-api.md#typescript-declarations))